### PR TITLE
do not use pool RWMutex in every function

### PR DESCRIPTION
**Pull request by HarrisChu**
1. change list to queue, no need to use RWMutex in every function.
2. enhance `getConnection` in initial pool

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+package nebula_go
+
+import (
+	"container/list"
+	"sync"
+)
+
+// Queue goroutine safe for list
+type Queue struct {
```
Comment (veezhang) [inline queue.go:15]: In computer science, a queue is a collection of entities that are maintained in a sequence and can be modified by the addition of entities at one end of the sequence and the removal of entities from the other end of the sequence.

Would it be better to use `SafeList`,`ConcurrentList` or other name?

refs: https://en.wikipedia.org/wiki/Queue_(abstract_data_type)

Comment (HarrisChu) [reply queue.go:15]: ok for me

File: connection_pool.go [hunk 2]
```diff
@@ -194,8 +196,6 @@ func (pool *ConnectionPool) Ping(host HostAddress, timeout time.Duration) error
 
 // Close closes all connection
 func (pool *ConnectionPool) Close() {
-	pool.rwLock.Lock()
```
Comment (veezhang) [inline connection_pool.go:197]: It's seems to be un-safe to iterate list, the length of list maybe changed during iterate.
```go
idleLen := pool.idleConnectionQueue.Len()
for i := 0; i < idleLen; i++ {
		pool.idleConnectionQueue.Front().Value.(*connection).close()
		pool.idleConnectionQueue.Remove(pool.idleConnectionQueue.Front())
	}
```

File: queue.go (hunk 1 above)
Comment (veezhang) [reply queue.go]: Other list iterations may also have the same problems.

File: connection_pool.go (hunk 2 above)
Comment (veezhang) [reply connection_pool.go:197]: Other list iterations may also have the same problems.

Comment (HarrisChu) [reply connection_pool.go:197]: agreed

**Comment by HarrisChu** [comment]
At first, just want to improve performance for `getIdleConn`.
But there's not much benefit by using concurrentList, as it should also have a lock for list iteration.

close the pr.

**Pull request closed without merging**